Emitter/Emitter.cs

```diff
@@ -52,6 +54,11 @@ public partial class Connection : IDisposable
         private readonly ReverseTrie<MessageHandler> Trie = new ReverseTrie<MessageHandler>(-1);
         private string DefaultKey = null;
 
+        private readonly ConcurrentDictionary<int, TaskCompletionSource<byte[]>> _waitingRequests = new ConcurrentDictionary<int, TaskCompletionSource<byte[]>>();
```

Comment:
Instead of using `_lowerCase` we should stick to the convention used in the library and declare these as `WaitingRequests`, `RequestNames` and `DefaultTimeout`. When used do `this.DefaultTimeout` etc.

Reply:
VS editor discourages using unnecessary `this`.